### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by toocomplex1**
added feature of image upload with article and search by tag

### CODE REVIEW

File: blog/app/views/users/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<h2>Log in</h2>
```
Comment (coolRaider) [inline blog/app/views/users/sessions/new.html.erb]: Please convert all .erb to .haml

File: blog/config/environments/development.rb [hunk 2]
```diff
@@ -35,6 +35,19 @@
...
   config.assets.raise_runtime_errors = true
+  
+  #added for uesr authentication features
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+  ActionMailer::Base.smtp_settings = 
+  {  
+   :address              => "smtp.gmail.com",  
+   :port                 => 587,  
+   :domain               => "gmail.com",  
+   :user_name            => "jaswantsinghyadav007@gmail.com",
```
Comment (coolRaider) [inline blog/config/environments/development.rb]: put the username and password in environment variables

File: blog/app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,7 +2,11 @@ class ApplicationController < ActionController::Base
 
 	
   http_basic_authenticate_with name: "akash", password: "secret", except: [:index, :show]
```
Comment (alienware) [inline blog/app/controllers/application_controller.rb]: Time to remove this then perhaps? :D

File: blog/app/models/article.rb [hunk 4]
```diff
@@ -10,3 +26,10 @@ def should_generate_new_friendly_id?
   validates :title, presence: true,
                     length: { minimum: 5 }
 end
+
+#for adding tags
+class Article < ActiveRecord::Base
```
Comment (alienware) [inline blog/app/models/article.rb:32]: WTH, one class, one definition. Don't separate just to keep versions of code.

File: blog/app/models/model.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Model < ActiveRecord::Base
```
Comment (alienware) [inline blog/app/models/model.rb:1]: Where required?

File: blog/app/views/devise/confirmations/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
```
Comment (alienware) [inline blog/app/views/devise/confirmations/new.html.erb]: Lets see haml-rails gem be properly integrated. It that doesn't work, discuss amongst all and come up with the solution.

**Comment by alienware** [comment]
Why such long PR, it even has devise integration. Couldn't understand.

**Pull request merged**